treewide: remove unused/redundant arguments, format workspace module, fix whitespace, remove `with lib;`

SigmaSquadron opened the pull request:
Various code refactoring changes. We may want to make use of Nixpkgs' EditorConfig settings/CI for this repo as well.

One of the commits is marked as `NEEDS DISCUSSION`. Replacing declared, yet unused arguments with `...` is frowned upon by some, but others prefer having no unused args. Let me know if that commit should be deleted.

SigmaSquadron committed:
refactor(treewide): remove unused expressions

Removes an useless `rec`, a few unused NixOS args, and a toString
inherit from builtins that is no longer used.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
chore(treewide): remove redundant parenthesis

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
chore(treewide): remove trailing whitespace

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
refactor(widgets): remove unused arguments and add `...`

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
style(modules/workspace): format with nixfmt-rfc-style

Looks like this file wasn't formatted during the tree-wide.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
refactor(input): remove `with lib;`

Fixes (partially) #380

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron committed:
refactor(kwin): remove `with lib;`

Fixes (partially) #380

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

HeitorAugustoLN commented on modules/input.nix:
```diff
@@ -433,77 +431,77 @@ in
         "altwin:meta_alt"
         "caps:shift"
-        "custom:types"
+        "custom:lib.types"
```
This shouldn't change.
```suggestion
        "custom:types"
```

SigmaSquadron replied:
Whoops! `rg` was a bit too eager there. Resolved.

HeitorAugustoLN commented:
Everything else looks good. I really liked the change from the unused args to `...`

SigmaSquadron committed:
docs(treewide): normalise URLs

- Removes the useless `?tab=readme-ov-file` from GitHub README URLs.
- Always points to KDE Invent instead of the GitHub mirror, and uses
tags instead of blobs.
- Changes the `lib.attrsets.mapAttrs'` manual page from Ryan's outdated
one to the new and bespoke Noogle documentation.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron commented:
Changes were addressed, the commits were renamed per convention, and there's a new commit to modernise/normalise the URLs in the documentation/code commits.

I wasn't exactly sure how to handle your merge commit though. I removed it and rebased manually.

HeitorAugustoLN commented:
Thanks for your contribution again :)
